### PR TITLE
/app/views/shared/_stats.html.erb refactor

### PR DESCRIPTION
**Pull request by OPARA-PROSPER**
# Description 

I refactored the _stats.html.erb partial view like @julianguyen suggested on issue #968
## More Details

Refactoring this partial view involved:
- Creating another partial view in the _shared folder_ i named the view __focus.html.erb__ 
- Replacing local_assigns[:data_type] == 'category' || local_assigns[:data_type] == 'mood' || local_assigns[:data_type] == 'strategy' with %w(category mood strategy).include? local_assigns[:data_type] 
## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/968
# Screenshots

# Test Coverage

✅ 
🚫

### CODE REVIEW

File: app/views/shared/_stats.html.erb [hunk 1]
```diff
@@ -17,29 +17,7 @@
...
-              <% elsif local_assigns[:data_type] == 'strategy' && (!local_assigns[:profile] || local_assigns[:profile] == current_user.id) %>
-                <%= viewers_hover(get_viewers_for(Strategy.where(id: key).first, 'strategy'), Strategy.where(id: key).first) %>
-              <% elsif local_assigns[:data_type] == 'strategy' && local_assigns[:profile] %>
-                <div class="someFocusTextName">
-                  <%= Strategy.where(id: key).first.name %>
-                </div>
-              <% end %>
-            </div>
-          </div>
+          <%= render 'shared/focus' %>
```
Comment (julianguyen) [inline app/views/shared/_stats.html.erb]: Does the page render properly out of curiosity?
In our codebase, we've been doing `<%= render partial: '/shared/stats', locals: { data_type: 'category' } %>` for example.

File: app/views/shared/_focus.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<div class="someFocus">
+    <div class="someFocusText">
+      <% if local_assigns[:data_type] == 'category' && (!local_assigns[:profile] || local_assigns[:profile] =current_user.id) %>
```
Comment (julianguyen) [inline app/views/shared/_focus.html.erb]: `local_assigns[:profile] =current_user.id` should be `local_assigns[:profile] == current_user.id`

Looks like this an issue in multiple places.

Comment (OPARA-PROSPER) [reply app/views/shared/_focus.html.erb]: Yah, just noticed.. I fixed this

File: app/views/shared/_stats.html.erb (hunk 1 above)
Comment (julianguyen) [reply app/views/shared/_stats.html.erb]: You should really be running the app locally on your machine while making changes to it. Unfortunately, we don't have automated tests for everything. Things like UI often times have to be tested manually.

Are you using Windows or Linux? I recommend using Linux and following our [Installation](https://github.com/ifmeorg/ifme/wiki/Installation) steps for that.

Comment (OPARA-PROSPER) [reply app/views/shared/_stats.html.erb]: I'm using a windows, still working to get a dual boot (Really need to keep using windows for now), Once i get it i'd be sure to install ruby on rails

Comment (julianguyen) [reply app/views/shared/_stats.html.erb]: Sounds good! I also recommend using Linux in VirtualBox. It takes a lot less time than doing a dual boot installation.

Setting up Ruby on Rails on Windows is possible but not as straightforward compared to Linux or macOS.

File: app/views/shared/_focus.html.erb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<div class="someFocus">
+    <div class="someFocusText">
+      <% if local_assigns[:data_type] == 'category' && (!local_assigns[:profile] || local_assigns[:profile] ==current_user.id) %>
+        <%= viewers_hover(get_viewers_for(Category.where(id: key).first, 'category'), Category.where(id: ke.first) %>
```
Comment (jm96441n) [inline app/views/shared/_focus.html.erb]: Can we move these AR queries into the helper method?

**Comment by jm96441n** [comment]
Can we move the AR queries that are present in the view into the controller? Running AR queries in the view breaks the MVC pattern (where the View is only responsible for displaying objects queried from the controller)

**Comment by julianguyen** [comment]
@OPARA-PROSPER Hey, I saw that you were setting up your dev environment! Any luck and are you still working on this issue?

**Pull request closed without merging**